packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -107,6 +111,114 @@ Example usage of the `nimble-rich-text-editor` in the application layer is as fo
+   these elements would be used:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-rich-text-mention-user>
+        <nimble-mapping-mention
+            key="user-id-1"
+            text="John Doe"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="user-id-2"
+            text="Alice Smith"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="user-id-3"
+            text="Bob Jones"
+        ></nimble-mapping-mention>
+    </nimble-rich-text-mention-user>
+
+    <nimble-rich-text-mention-issue>
+        <nimble-mapping-mention
+            key="issue-id-1"
+            text="Spec for rich text editor"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="issue-id-2"
+            text="Mention support in rich text components"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="issue-id-3"
+            text="Issue in pasting a link"
+        ></nimble-mapping-mention>
+    </nimble-rich-text-mention-issue>
+</nimble-rich-text-editor>
+```
+
+_Alternatives_:
+
+Slot based approach: All user information will be communicated through `nimble-list-option` elements. These elements will
+contain the user's ID in their `value` attribute and the user's name in their text content. When the **"@"** symbol is
+inserted into the editor, these elements will be presented as a dropdown list.
+
+Pros:
+
+1. When the client does not specify a `value` attribute, the options are sent without a user's ID, and in such cases,
+   the value attribute is derived from the content of the option.
+
+Cons:
+
+1. It is impossible to have multiple type of mentions when `nimble-list-option` is used in the default slot. If we use
+   slot names to differentiate between various type of mentions, the rich text editor needs to be aware of all the type
+   of mentions like user slot name, issue slot name etc, through an attribute value.
+2. The slot-based approach is not configurable, meaning that if a user wishes to customize the entire user mention with
+   some arbitrary value, the only viable option is to include it within the options. In contrast, the configuration-based
+   approach readily allows for the inclusion of arbitrary values in the parent container element, making customization more
+   straightforward.
+3. The UI of the list option is completely dependent on the client application.
+4. All the keyboard and mouse interactions within the dropdown list should be handled internally within the editor as `nimble-list-option`
+   does not support any.
+
+#### Client Usage Guidance on Filtered Users:
+
+Initially, the list of users can be sorted in alphabetical order based on their usernames. Then, within the `nimble-mapping-mention`,
+only the initial 20 user details should be transmitted. These details will be used to establish the key-value pairs, consisting of user ID and username.
```

Comment:
@m-akinc Since this is based on the filtering that happens in the client application, we will be unaware if the client provides "Only first 20" or if the 20 is actually every option available after filtering. So can this be deferred to client scope and expose an attribute named `mention-filter-info`: a string that gets the message from the client and displays it as per the UX designer's suggestion in the popup?

The message can be constructed in the client application based on whether they provide the first 20 after filtering and display the same in the popup.  For example, if we type `@ab` and the client provides just the first 20 users starting with `ab`, which they can compare the actual length with the current length and give us the message `only first 20 shown`.

If there is no message provided from the client, we can ignore it by not showing any info messages. Also, here the message can be localized in the client application instead of handling it within the nimble components.

Reply:
Yes, I would probably recommend a slot at the end that either allows the end-user to load more than 20. A read-only message similar to what the query builder bar is currently doing would also work. Something like "Showing 20 matching names"? I don't love this solution for either component in the long run, but there's already a precedent for it in the product.

Reply:
Thanks for the responses. We could consider that the "Load more" is out of scope for now as it adds more weight to the initial basic mentioning support.

> Or we could simply not show anything and assume the user will continue typing if the person they're trying to find hasn't yet shown up in the dropdown.

As per the above suggestion, we will just leave it as-is and make the user type the name until they find it. As I mentioned in the above comment, it is again letting us create a new API to pass the message for the read-only message like "Showing 20 matching users" and not sure of its value.

Reply:
OK. Let's skip passing a message; I don't think it's necessary.